### PR TITLE
[RSDK-1963] servo digital interrupts are only in the raspberry pi code

### PR DESCRIPTION
**Pull request by penguinland**
Github thinks @martha-johnston should review this, so I'll assign to her instead of using the randomizer.

I had this much, much cleaner this morning, where the only part left in the `board` package was the interface, and all the implementation details were left up to the boards. but then it turned out that a bunch of unit tests were using these as well, and I had to put a bunch of stuff back.

Main changes:
- The `PostProcessor` type is defined in board/digital_interrupts.go, rather than board/board.go. 
- There is a copy of all the digital interrupt stuff (including the configs) in board/pi/impl/ now. That component uses this new copy.
- The main (non-pi) version of digital interrupts no longer has a `ServoDigitalInterrupt` implementation.
- The main (non-pi) version of digital interrupt configs no longer takes a Type or Formula.

Part of me wants to go further and remove the `PostProcessor` from everything except the client-side SDK because it's unused everywhere else. but that would probably require a scope doc because it's changing the public interface.

### CODE REVIEW

File: components/board/digital_interrupts.go
Comment (martha-johnston) [inline components/board/digital_interrupts.go:105]: is this just one of those unused functions you mentioned and that's why it's removed?

**Review by martha-johnston — APPROVED** [review]
lgtm assuming no behavior changes (which I'm assuming you tested). Maybe you can ask someone with more opinions than me if those extra functions that are only being used in unit tests are necessary, or if the functions and tests can just both be removed. some like `Tick` might be necessary since it's used for encoder tests, but maybe there are some that aren't?

File: components/board/digital_interrupts.go
Comment (penguinland) [reply components/board/digital_interrupts.go:105]: Yup, this is never used.

**Comment by penguinland** [comment]
> extra functions that are only being used in unit tests

The only remaining function that I think isn't absolutely necessary is `AddPostProcessor`, and removing that requires changing the publicly available functions in the SDK, which will require a scope doc. :exploding_head: but maybe I can remove it from the RDK-internal parts, and just leave it in the SDK section! I'll think about that for a follow-up PR.

The only remaining function used in tests is `Tick`, and I don't see how to make tests without it. The part that caught me off-guard is that so many tests were using these at all: we have mock versions in `testutils/inject/digital_interrupt.go`, but half the tests that need interrupts are using the non-mock ones. Maybe I should make a follow-up PR to change those? but it'll be a hassle for frankly not much improvement.

If you meant that maybe we can remove these from the rpi implementation and just have the one implementation everywhere, we probably can't do that for legacy reasons. Back in the day people built various things that require servo digital interrupts (e.g., the anemometer in Peter L's weather station), and at least for the moment it's easier to just put up with them than to convince everyone we can remove them. Hopefully that'll change when Olivia has the scope doc for a digital interrupt API.

File: components/board/pi/impl/digital_interrupts.go
Comment (randhid) [inline components/board/pi/impl/digital_interrupts.go:270]: Do these need to be exported functions at all anymore?

Comment (penguinland) [reply components/board/pi/impl/digital_interrupts.go:270]: Hm... I'm leaning towards "yes," but am not confident of that. This one is public because that's part of the `board.DigitalInterrupt` interface. It's used publicly by the fake board implementation, and it wouldn't surprise me if users of the Go SDK relied on that part of the interface, too. but maybe it could be made private and removed from the interface, and it's just a coincidence that 2 of the 3 implementations of the interface have these?

I'm inclined to just merge this as-is and think that through more another time, but I'm open to thinking it through now if you prefer.

Comment (randhid) [reply components/board/pi/impl/digital_interrupts.go]: Add a todo - unexport AddPostProcessor function from `pi/impl/digital_intterupts.go` and a jira ticket for further clean up if you still prefer doing that.

**Comment by penguinland** [comment]
Take one final look before I merge...

File: components/board/pi/impl/digital_interrupts.go
Comment (penguinland) [reply components/board/pi/impl/digital_interrupts.go]: I looked into putting it back, and realized it would be easier to just roll forward and remove it entirely. I still wish I'd broken this up into 2 or perhaps 3 smaller PRs.

**Pull request merged**
